apps/opik-python-backend/src/opik_backend/studio/types.py

```diff
@@ -0,0 +1,122 @@
+"""Data types and context objects for Optimization Studio."""
+
+from dataclasses import dataclass
+from typing import Dict, Any, Optional, List
+from uuid import UUID
+
+
+@dataclass
```

Comment:
if it's something that is going to reflect a contract between the client and Python Backend or Python Backend and Java Backend (or smth like that), I recommend using pydantic instead of dataclasses.

Pydantic has strong data validation mechanisms. Dataclasses don't validate anything, they just provide a more convenient way of aggregating data than dicts.